pkg/router/controller/host_admitter.go

```diff
@@ -90,12 +91,13 @@ type HostAdmitter struct {
 	claimedHosts     RouteMap
 	claimedWildcards RouteMap
 	blockedWildcards RouteMap
+	labels           labels.Selector
 }
 
 // NewHostAdmitter creates a plugin wrapper that checks whether or not to
 // admit routes and relay them to the next plugin in the chain.
 // Recorder is an interface for indicating why a route was rejected.
-func NewHostAdmitter(plugin router.Plugin, fn RouteAdmissionFunc, allowWildcards, disableNamespaceCheck bool, recorder RejectionRecorder) *HostAdmitter {
+func NewHostAdmitter(plugin router.Plugin, fn RouteAdmissionFunc, allowWildcards, disableNamespaceCheck bool, recorder RejectionRecorder, NamespaceLabels labels.Selector) *HostAdmitter {
```

Comment:
Do we want to pass the labels?   Or just make the interface be isSharded bool?

Reply:
i'll add the check that @pravisankar suggested below